### PR TITLE
Add longhorn 1.2.0 chart dev 2.5

### PR DESCRIPTION
**Pull request by innobead**
Signed-off-by: David Ko <dko@suse.com>

This PR is ready for review, but the images will be finalized until the GA release of v1.2.0, so it is marked as a draft for now.

cc @khushboo-rancher @meldafrawi @shuo-wu

### CODE REVIEW

**Comment by khushboo-rancher** [comment]
Validated with this branch.

Validation - **Pass**

File: index.yaml
Comment (shuo-wu) [inline index.yaml:444]: Can we modify the version for existing entries? Similar below
Need to wait for Rancher team's review.

File: index.yaml
Comment (shuo-wu) [inline index.yaml:703]: I think we need to add a new entry rather than modifying the existing one.

File: index.yaml
Comment (innobead) [reply index.yaml:444]: This is automatically generated. 

@jiaqiluo any idea for this or what else we need to change in package.yaml?

Comment (innobead) [reply index.yaml:444]: cc @aiyengar2

**Comment by innobead** [comment]
Before merging, we need QA to verify again before merging. cc @khushboo-rancher

File: index.yaml
Comment (jiaqiluo) [reply index.yaml:702]: see my previous comment.

File: index.yaml
Comment (jiaqiluo) [reply index.yaml:444]: The items are ordered from the largest version to the smallest, so we see those changes here.
the real changes are the additions of longhorn 1.2.0 and longhorn-crd 1.2.0, which is expected.

**Review by jiaqiluo — APPROVED** [review]
LGTM.

**Comment by khushboo-rancher** [comment]
Validated the chart - LGTM

**Comment by innobead** [comment]
> Validated the chart - LGTM
> 
> The issue [longhorn/longhorn#2964](https://github.com/longhorn/longhorn/issues/2964) is solved with this PR
> 
> @innobead are the below versions as expected?
> Screen Shot 2021-09-07 at 3 11 37 PM

I think yes, but need @jiaqiluo to confirm again.

**Comment by jiaqiluo** [comment]
I am sure that `1.1.100+up1.1.1` should be removed as it is a duplication of `1.1.100` (we plan to trim all those unneeded charts before releasing 2.5.10)

But I am not sure about `1.1.200+up1.1.2` @innobead  Do we have a plan to release it for 2.5.10? 

If not, could you remove both of them(assets, packages, and entries in the index.yaml)?

**Comment by innobead** [comment]
> I am sure that `1.1.100+up1.1.1` should be removed as it is a duplication of `1.1.100` (we plan to trim all those unneeded charts before releasing 2.5.10)
> 
> But I am not sure about `1.1.200+up1.1.2` @innobead Do we have a plan to release it for 2.5.10?
> 
> If not, could you remove both of them(assets, packages, and entries in the index.yaml)?

We should have 1.1.2 because it's the patch release released.

**Comment by jiaqiluo** [comment]
> > I am sure that `1.1.100+up1.1.1` should be removed as it is a duplication of `1.1.100` (we plan to trim all those unneeded charts before releasing 2.5.10)
> > But I am not sure about `1.1.200+up1.1.2` @innobead Do we have a plan to release it for 2.5.10?
> > If not, could you remove both of them(assets, packages, and entries in the index.yaml)?
> 
> We should have 1.1.2 because it's the patch release already released.

Oh, ok. I just found that it was released as 100.0.0+up1.1.2 in the [release-v2.6 branch ](https://github.com/rancher/charts/tree/release-v2.6/charts/longhorn/longhorn). Let's keep `1.1.200+up1.1.2` in this branch.

**Review by aiyengar2 — DISMISSED** [review]
LGTM

**Comment by aiyengar2** [comment]
FYI I have the same comment on this as I did on https://github.com/rancher/charts/pull/1428#pullrequestreview-749556036. If you plan to maintain two release lines for longhorn in 2.5, it would be better to split the `packages/` entries into two, as described in the comment.

File: packages/longhorn/generated-changes/patch/questions.yml.patch
Comment (aiyengar2) [inline packages/longhorn/generated-changes/patch/questions.yml.patch]: Is this in the wrong commit?

File: packages/longhorn/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/longhorn/generated-changes/patch/values.yaml.patch]: Is this in the wrong commit?

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
Need to delete `charts/`, `assets/`, and `index.yaml` entries for the following orphaned charts, since they have been relocated to their own packages:

- [ ] longhorn-crd 1.1.100+up1.1.1
- [ ] longhorn-crd 1.1.200+up1.1.2
- [ ] longhorn 1.1.100+up1.1.1
- [ ] longhorn 1.1.200+up1.1.2

Can you add this to a separate commit post make charts?

**Comment by innobead** [comment]
> Need to delete `charts/`, `assets/`, and `index.yaml` entries for the following orphaned charts, since they have been relocated to their own packages:
> 
> * [ ]  longhorn-crd 1.1.100+up1.1.1
> * [ ]  longhorn-crd 1.1.200+up1.1.2
> * [ ]  longhorn 1.1.100+up1.1.1
> * [ ]  longhorn 1.1.200+up1.1.2
> 
> Can you add this to a separate commit post make charts?

Please note that after removing` longhorn 1.1.200+up1.1.2` from the old longhorn package, the `index.yaml` required to update to make 1.1.2 release point to the new location `assets/longhorn-1.1/longhorn-crd-1.1.200+up1.1.2.tgz`.

Because 1.1.2 chart was introduced before (https://github.com/rancher/charts/pull/1340) even it's not released yet (release-2.5), we need to revalidate 1.1.2 as per this change.

**Comment by aiyengar2** [comment]
> > Need to delete `charts/`, `assets/`, and `index.yaml` entries for the following orphaned charts, since they have been relocated to their own packages:
> > 
> > * [ ]   longhorn-crd 1.1.100+up1.1.1
> > * [ ]   longhorn-crd 1.1.200+up1.1.2
> > * [ ]   longhorn 1.1.100+up1.1.1
> > * [ ]   longhorn 1.1.200+up1.1.2
> > 
> > Can you add this to a separate commit post make charts?
> 
> Please note that after removing` longhorn 1.1.200+up1.1.2` from the old longhorn package, the `index.yaml` required to update to make 1.1.2 release point to the new location `assets/longhorn-1.1/longhorn-crd-1.1.200+up1.1.2.tgz`.
> 
> Because 1.1.2 chart was introduced before (#1340) even it's not released yet (release-2.5), we need to revalidate 1.1.2 as per this change. cc @khushboo-rancher

yes, we need to ensure the index.yaml at the end contains the new chart in the new asset location. 

One way to do this would be to add the above changes to the commit **before** make charts instead of after; then, when you run make charts, it will automatically detect that there is no chart entry for the asset and add it to the index.yaml

**Review by aiyengar2 — DISMISSED** [review]
LGTM

**Comment by khushboo-rancher** [comment]
Revalidated the chart with v1.1.2 and v1.2.0 with this branch - **LGTM**

Screen Shot 2021-09-09 at 4 42 33 PM

**Review by cbron — DISMISSED** [review]
lgtm, the versions look weird but thats ok, it makes sense why it ended up that way.

File: index.yaml
Comment (jiaqiluo) [inline index.yaml:402]: The changes in this file are actually:
- add the 1.2.0 
- remove the 1.1.100+up1.1.1

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
Resolved the index.yaml conflict. 

@cbron Please help review again. Thanks.

**Review by jiaqiluo — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Comment by khushboo-rancher** [comment]
Revalidated the chart with this branch - **LGTM**

Verified installation, uninstallation, upgrade (v1.1.2- v1.2.0) and attaching/detaching, mounting of rwo and rwx volume.

cc: @innobead

**Comment by innobead** [comment]
@cbron @jiaqiluo QA already verified it, so please feel free to merge. Thanks.

**Pull request merged**
